src/main/java/ru/ayaz/UserMessageDistributor.java

```diff
@@ -0,0 +1,63 @@
+package ru.ayaz;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import static ru.ayaz.MessageParser.isCommand;
+
+public class UserMessageDistributor implements Runnable {
+
+    private Map<String, User> userMap;
+    private Map<String, UserSocketHandler> userSocketHandlerMap;
+    private BlockingQueue<UserMessage> messageQueue;
+    private MessageBroadcaster broadcaster;
+    private CommandExecutor executor;
+
+    UserMessageDistributor(HashMap<String, User> userMap, HashMap<String, UserSocketHandler> userSocketHandlerMap,
```

Comment:
&#x1F34E; **Issue**
The way the map of users / user handlers is created outside an object and injected into several of them feels like some Entity in your Domain model was brutally murdered, chopped into several pieces and its guts were pulled out for everyone to see ...
You should put it back into one of domain entities and encapsulate it behind methods like "addUserToChatRoom", "kickUserOutOfChatRoom", etc.